pkg/api/v1/types.go

```diff
@@ -862,6 +863,8 @@ type PodSpec struct {
 	DNSPolicy DNSPolicy `json:"dnsPolicy,omitempty" description:"DNS policy for containers within the pod; one of 'ClusterFirst' or 'Default'"`
 	// NodeSelector is a selector which must be true for the pod to fit on a node
 	NodeSelector map[string]string `json:"nodeSelector,omitempty" description:"selector which must match a node's labels for the pod to be scheduled on that node"`
+	// AntiAffinitySelectors are selectors which must not match against existing pods labels for the pod to fit on a node
+	AntiAffinitySelectors []labels.LabelSelector `json:"antiAffinitySelectors,omitempty" description:"list of selectors which must not match against existing pods labels for the pod to fit on a node"`
```

Comment:
I would suggest calling this NodeConflictSelectors

We're already using "affinity" to refer to best-effort spreading policies, so I think it's better if we use a new word ("conflict") here. I believe @bgrant0607 was OK with this.

Also, the reason I'm suggesting NodeConflictSelectors instead of ConflictSelectors is because we might eventually want Pods to be able to conflict with Pods that are already running on the node, not just with labels on the Node (in which case we might call that PodConflictSelectors or something like that).